### PR TITLE
{bio}[GCCcore/9.3.0] MUMmer v3.23

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node371.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/17fd414c8161fb9c789b8262482782cc for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2602.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/7bce7ee8c408303adc835b2dd329f754 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node124.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/a717ffd474e3cf736b79a52b09628147 for a full test report.

**Review by lexming — DISMISSED** [review]
LGTM

**Comment by lexming** [comment]
@boegel there are issues in CentOS 8 related to UTF-8, is that a blocker for you?

**Comment by boegel** [comment]
@lexming That actually signals a bug in the MUMmer easyblock which is also relevant fro EasyBuild framework (because the MUMmer easyblock should be using `apply_regex_substitutions`, but that currently suffers from the same problem).

I'll look into fixing this, lets keep this open as a reminder...

**Comment by boegel** [comment]
Just to expose the problem a bit more: fails when running EasyBuild on top of Python 3 with:

```
ERROR: Traceback (most recent call last):
  File "/lib/python2.7/site-packages/easybuild_framework-4.3.1.dev0-py2.7.egg/easybuild/main.py", line 115, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "/lib/python2.7/site-packages/easybuild_framework-4.3.1.dev0-py2.7.egg/easybuild/framework/easyblock.py", line 3301, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/lib/python2.7/site-packages/easybuild_framework-4.3.1.dev0-py2.7.egg/easybuild/framework/easyblock.py", line 3204, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/lib/python2.7/site-packages/easybuild_framework-4.3.1.dev0-py2.7.egg/easybuild/framework/easyblock.py", line 3059, in run_step
    step_method(self)()
  File "/lib/python2.7/site-packages/easybuild_easyblocks-4.3.1.dev0-py2.7.egg/easybuild/easyblocks/m/mummer.py", line 79, in install_step
    for line in fileinput.input(fil, inplace=1, backup='.orig.eb'):
  File "/usr/lib64/python3.6/fileinput.py", line 250, in __next__
    line = self._readline()
  File "/usr/lib64/python3.6/fileinput.py", line 364, in _readline
    return self._readline()
  File "/usr/lib64/python3.6/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa4 in position 24: invalid start byte
```

**Comment by boegel** [comment]
@lexming Framework issue fixed in https://github.com/easybuilders/easybuild-framework/pull/3450, MUMmer easyblock updated to leverage `apply_regex_substitutions` and fix the issue that arose here when running EasyBuild on top of Python 3.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2636.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/558586d1a6ae2d331fbb2b2aad350747 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
nuc.lan - Linux Fedora 32, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.8.5
See https://gist.github.com/c0872f2ab0966b7b5de07006a263bbaa for a full test report.

File: easybuild/easyconfigs/m/MUMmer/MUMmer-3.23-GCCcore-9.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Copyright:: Copyright 2012-2014 Uni.Lu/LCSB, NTUA
+# Authors::   Cedric Laczny <cedric.laczny@uni.lu>, Fotis Georgatos <fotis@cern.ch>
+# License::   MIT/GPL
+# $Id$
+#
+# This work implements a part of the HPCBIOS project and is a component of the policy:
+# http://hpcbios.readthedocs.org/en/latest/HPCBIOS_2012-94.html
+##
+
+# easyblock = mummer.py
```
Comment (verdurin) [inline easybuild/easyconfigs/m/MUMmer/MUMmer-3.23-GCCcore-9.3.0.eb:13]: @boegel please re-instate the previous credits, old though they may be...

**Review by verdurin — CHANGES_REQUESTED** [review]

**Comment by zao** [comment]
Test report by @zao
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
freja - Linux Ubuntu 20.04, x86_64, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 3.8.2
See https://gist.github.com/0353207977306b5b72a595b957590906 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3401.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/6307ed92581f9d0ddb4d070123fa8325 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 0 out of 0 (1 easyconfigs in this PR)
nuc.lan - Linux Fedora 32, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.8.5
See https://gist.github.com/2e9a46361a5e1e22c636e51c784e64a8 for a full test report.

**Review by verdurin — APPROVED** [review]
Looks fine.

**Pull request merged**